### PR TITLE
[4.0] Fixed the parent item dropdown

### PR DESCRIPTION
**Pull request by fatmaakay**
Pull Request for Issue #21225  .

### Summary of Changes
Parent Item was always empty after changing the menu.
image

This PR fixes this problem.

### Testing Instructions
1. Click on system 
2. Click on global configuration
3. Click on system
4. Click on debug system to make it active
5. Save and close  
6. Create a new menu.
7. Open the menu item.
8.  Click on the menu.
image
9. Check that the parent item is empty
image
10. Apply patch
11. Change the menu item again
12. Check that the parent item is not empty
image

### Expected result
Expected result is the parent item is not empty

### Actual result
Actual result is parent item is empty

### Documentation Changes Required
None

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on e75c3e34971c004c35dc1e135ece37132a10a975<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21348">issues.joomla.org/tracker/joomla-cms/21348</a>.</sub>

**Comment by roland-d** [comment]
@infograf768 Could you test this as well please?

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 8988c5d79998dc1f3759b5640b2ad84ef911d150<br /><br />Needs to use npm install as node build.js --compile-js is not enough here. (my test site contains https://github.com/joomla/joomla-cms/pull/21204

But it works correctly.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21348">issues.joomla.org/tracker/joomla-cms/21348</a>.</sub>

**Comment by infograf768** [comment]
Setting this to RTC as it can't be tested by people not using npm...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21348">issues.joomla.org/tracker/joomla-cms/21348</a>.</sub>

File: build/media/com_menus/js/admin-item-edit.js [hunk 1]
```diff
@@ -47,18 +47,21 @@
...
 						var option       = document.createElement('option');
-						option.innerText = value;
+						option.innerText = val.title;
 						option.id        = val.id;
 						document.getElementById('jform_parent_id').appendChild(option);
 					});
-					document.getElementById('jform_parent_id').trigger('change');
+
+                    var event = document.createEvent('HTMLEvents');
+                    event.initEvent('change', true, false);
```
Comment (infograf768) [inline build/media/com_menus/js/admin-item-edit.js:63]: To normalize, would be good to indent with tabs for the 2 above lines instead of spaces

**Pull request merged**